pkg/importer/http-datasource.go

```diff
@@ -154,6 +154,9 @@ func (hs *HTTPDataSource) Transfer(path string) (ProcessingPhase, error) {
 			return ProcessingPhaseError, err
 		}
 		size, err := util.GetAvailableSpace(path)
+		if err != nil {
```

Comment:
I think this is why tests fail, we rely on `ErrInvalidPath` to tell us scratch space is needed

Reply:
But this value assignment of the `err` var is never checked. Maybe there is a hidden bug here that this check finds.

Reply:
I mean we should definitely keep the err check, just need to find a way to return `ErrInvalidPath` when needed
since we rely on it https://github.com/kubevirt/containerized-data-importer/blob/aef617e7c9de2c641dc949cd8109d23231406cf6/pkg/importer/http-datasource.go#L160-L163

Reply:
got you. and since `GetAvailableSpace` returns `-1` in case of error, you get the same behaviour. What do you think about combining the conditions? e.g.
```suggestion
		size, err := util.GetAvailableSpace(path)
		if err != nil || size <= 0 {
			//Path provided is invalid.
			return ProcessingPhaseError, ErrInvalidPath
		}
```

Reply:
I'm not sure. Maybe you can tidy that later, but this will not change the current behavior. https://github.com/kubevirt/containerized-data-importer/blob/f9c049f0e481d4697dfa9444db8aa64620a17e91/pkg/util/util.go#L102-L109

For any error in `GetAvailableSpace`, the function returns -1, so the the old `if size <= int64(0)` check, bahaved the same.

Reply:
Yep right